File: BBIS.Application/DTOs/DonorScreening/BBInventoryDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BBIS.Application.DTOs.DonorScreening
{
    public class BBInventoryDTO
    {
        public string bag_no { get; set; } = string.Empty;         // required|string
        public string reg_no { get; set; } = string.Empty;         // required|string
        public string blood_group { get; set; } = string.Empty;    // required|string
        public string component { get; set; } = string.Empty;     // required|string
        public string volume { get; set; } = string.Empty;     // required|string|max:25
        public DateTime? collection_date { get; set; }             // nullable|date
        public DateTime expiry { get; set; }                      // required|date
        public string source { get; set; } = string.Empty;        // required|string
        public string? bgroup { get; set; }                       // nullable|string
        public string? rh { get; set; }                           // nullable|string
        public string? bp_statusid { get; set; }                   // nullable|string
        public bool? has_allocation { get; set; }                  // nullable|boolean
        public string? firstname_alloc { get; set; }               // nullable|string
        public string? middlename_alloc { get; set; }              // nullable|string
        public string? lastname_alloc { get; set; }                // nullable|string
        public string? segmentserialno { get; set; }              // nullable|string
    }
}

File: BBIS.Api/ClientApp/src/models/DonorScreening/BBInventoryDTO.ts
export interface IBBInventory {
    bagNo: string;                
    regNo: string;                
    bloodGroup: string;           
    component: string;            
    volume: string;            
    collectionDate?: string;       
    expiry: string;                
    source: string;               
    bGroup?: string;              
    rh?: string;                  
    bpStatusId?: string;          
    hasAllocation?: boolean;      
    firstnameAlloc?: string;      
    middlenameAlloc?: string;     
    lastnameAlloc?: string;       
    segmentSerialNo?: string;     
}

export class BBInventoryDTO implements IBBInventory {
    bagNo = '';
    regNo = '';
    bloodGroup = '';
    component = '';
    volume = '';
    collectionDate?: string;
    expiry = '';
    source = '';
    bGroup?: string;
    rh?: string;
    bpStatusId?: string;
    hasAllocation?: boolean;
    firstnameAlloc?: string;
    middlenameAlloc?: string;
    lastnameAlloc?: string;
    segmentSerialNo?: string;
}
